### PR TITLE
Update release scripts for 0.5.0 release

### LINKED ISSUE #6: GH_TOKEN should not be needed to run `generate_releaes_notes.py -h`
```pytb
$ python generate_release_notes.py -h
Traceback (most recent call last):
  File "/Users/jni/projects/napari-release-tools/generate_release_notes.py", line 32, in <module>
    from release_utils import (
  File "/Users/jni/projects/napari-release-tools/release_utils.py", line 68, in <module>
    raise RuntimeError(
RuntimeError: It is necessary that the environment variable `GH_TOKEN` be set to avoid running into problems with rate limiting. One can be acquired at https://github.com/settings/tokens.

You do not need to select any permission boxes while generating the token.
```

### PR DESCRIPTION
**Pull request by Czaki**
Closes #6

### CODE REVIEW

File: find_contributors_without_citation.py [hunk 1]
```diff
@@ -30,6 +31,10 @@
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument("--milestone", help="The milestone to check")
+    parser.add_argument("--repo", help="The repo to check", default="napari/napari")
+    parser.add_argument(
+        "--additional-repo", help="The additional repo to check", action="append"
+    )
```
Comment (jni) [inline find_contributors_without_citation.py:37]: I would make repo a list rather, `nargs='+'`

Comment (jni) [reply find_contributors_without_citation.py:37]: https://docs.python.org/3/library/argparse.html#nargs

Comment (Czaki) [reply find_contributors_without_citation.py:37]: Action "append" allow to use flag multiple times.

Comment (jni) [reply find_contributors_without_citation.py:37]: ok then why not *just* have repo with append? Why a second arg?

Comment (Czaki) [reply find_contributors_without_citation.py:37]: To semantically point which repository is the main one. But maybe it is not the best solution.

Comment (Czaki) [reply find_contributors_without_citation.py:37]: For auto-fetching citation.cff file. But maybe auto-fetching is obsolete, and only local path should be supported?

Comment (jni) [reply find_contributors_without_citation.py:37]: I'm happy with any of:
- first repo is default
- check all repos
- add a separate `--citations-repo` arg
- use only a local file

**Comment by jni** [comment]
wanna undraft this one @Czaki? I'll merge since it's an improvement one way or another!

**Comment by Czaki** [comment]
> wanna undraft this one @Czaki? I'll merge since it's an improvement one way or another!

During previous releases, I just added changes until release happened, and then merged. I do not see strong benefit from early merge. But may add this fix to find contributors without citation.

**Comment by jni** [comment]
@Czaki note I've pushed some additional notes here. I've added a header as that can sometimes be updated — will integrate into the script later.

**Comment by jni** [comment]
@Czaki do you want to check this before merge? I hope to release 0.5.1 soon, maybe Monday. (I think I will release an rc but only to check that the pipelines are working, I would release the full release quickly after that, because there's not that many commits between releases so I think it's less critical to have a lot of testing.)

**Comment by jni** [comment]
@Czaki I don't know how to run black/ruff-fmt on this without having something new to commit

**Comment by Czaki** [comment]
don't know how to run black/ruff-fmt on this without having something new to commit

`pre-commit run --all`

**Review by Czaki — COMMENTED** [review]
look slike only change `"` on `'` in multiple files

**Pull request merged**